app/src/main/java/com/exallium/mvvmcleanexample/app/binding/TextInputEditTextBindings.kt

```diff
@@ -3,16 +3,20 @@ package com.exallium.mvvmcleanexample.app.binding
 import android.databinding.BindingAdapter
 import android.support.design.widget.TextInputEditText
 import com.jakewharton.rxbinding2.widget.textChanges
-import io.reactivex.Observable
-import io.reactivex.functions.Consumer
+import io.reactivex.disposables.Disposable
 
 object TextInputEditTextBindings {
 
-    interface StringConsumer : Consumer<Observable<String>>
+    interface StringConsumer {
+        fun invoke(text: CharSequence)
+    }
 
     @JvmStatic
-    @BindingAdapter("bindTextChanges")
-    fun setChangesObservable(textInputEditText: TextInputEditText, consumer: StringConsumer) {
-        consumer.accept(textInputEditText.textChanges().map(CharSequence::toString))
+    @BindingAdapter("onTextChanged")
+    fun onTextChanged(textInputEditText: TextInputEditText, consumer: StringConsumer?) {
+        (textInputEditText.tag as? Disposable)?.dispose()
```

Comment:
If there's a precedent of utilizing the tag for something like this in the binding framework, then I guess that's ok.  I'm curious though about applying multiple bindings to a single view, and whether you'd get clobbering.